src/lib/select/select.md

```diff
@@ -1,111 +1,166 @@
+
+<!-- example(select-panel-class) -->
+
+### Changing when error messages are shown
+
+The `<mat-form-field>` allows you to
+[associate error messages](https://material.angular.io/components/form-field/overview#error-messages)
+with your `<mat-select>`. By default, these error messages are shown when the control is invalid and
+either the user has interacted with (touched) the element or the parent form has been submitted. If
+you wish to override this behavior (e.g. to show the error as soon as the invalid control is dirty
+or when a parent form group is invalid), you can use the `errorStateMatcher` property of the
+`<mat-select>`. The property takes an instance of an `ErrorStateMatcher` object. An
+`ErrorStateMatcher` must implement a single method `isErrorState` which takes the `FormControl` for
+this `<mat-select>` as well as the parent form and returns a boolean indicating whether errors
+should be shown. (`true` indicating that they should be shown, and `false` indicating that they
+should not.)
+
+<!-- example(select-error-state-matcher) -->
+
+A global error state matcher can be specified by setting the `ErrorStateMatcher` provider. This
+applies to all inputs. For convenience, `ShowOnDirtyErrorStateMatcher` is available in order to
+globally cause input errors to show when the input is dirty and invalid.
 
 ```ts
 @NgModule({
   providers: [
-    {provide: MAT_PLACEHOLDER_GLOBAL_OPTIONS, useValue: { float: 'always' }}
+    {provide: ErrorStateMatcher, useClass: ShowOnDirtyErrorStateMatcher}
   ]
 })
 ```
 
-### Customizing the trigger label
-If you want to display a custom trigger label inside a select, you can use the `mat-select-trigger` element:
-
-```html
-<mat-form-field>
-  <mat-select placeholder="Favorite food" #select="matSelect">
-    <mat-select-trigger>You have selected: {{ select.selected?.viewValue }}</mat-select-trigger>
-    <mat-option *ngFor="let food of foods" [value]="food.value">{{ food.viewValue }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
-
-Here are the available global options:
-
-| Name            | Type    | Values              | Description                               |
-| --------------- | ------- | ------------------- | ----------------------------------------- |
-| float           | string  | auto, always, never | The default placeholder float behavior.   |
-
 ### Keyboard interaction
+
 - <kbd>DOWN_ARROW</kbd>: Focus next option
 - <kbd>UP_ARROW</kbd>: Focus previous option
 - <kbd>ENTER</kbd> or <kbd>SPACE</kbd>: Select focused item
 
 ### Accessibility
+
 The select component without text or label should be given a meaningful label via
 `aria-label` or `aria-labelledby`.
 
 The select component has `role="listbox"` and options inside select have `role="option"`.
+
+### Troubleshooting
+
+#### Error: Cannot change `multiple` mode of select after initialization
+
+This error is thrown if you attempt to bind the `multiple` property on `<mat-select>` to a dynamic
```

Comment:
Why can't we change this at run-time?

Reply:
I put that in place when I initially implemented multiple selection. It's there's some ambiguity about what to do with the selected values when you're switching modes. Also it's unlikely that you'll need to toggle it after init anyway (we haven't had any issues about it yet).